### DISCUSSION TITLE
V3 : cashed a id after remove item  and doesn't get the new  id item.

### DISCUSSION THREAD

**Discussion by rawezhcode** [livewire/livewire#5848] [7. Bugs (archived)] [answered]
https://github.com/livewire/livewire/assets/53620168/0d93944f-c63c-4183-aa6b-0044aef7263f

When I delete the pre-last item and then when I want to delete the last item, it still gets the old item id and doesn't get the new one.

**Comment by przemi** [comment]
Same problem here :(

**Comment by PhiloNL** [comment]
I suspect this is related to https://github.com/livewire/livewire/pull/5776

**Comment by rawezhcode** [comment, marked as answer]
I found the error that all lists should be wire:key when using foreach

```
<div>
    <ul class="w-[230px] divide-y divide-gray-200 dark:divide-gray-700">
        @foreach ($carts as $item)         
        <li class="pb-3 sm:pb-4" wire:key="item-{{ $item->id }}">
            <div class="flex items-center space-x-4">
                <button wire:click="remove({{ $item->id }})" type="button" class="w-[30px] h-[30px] rounded-full z-[999]">
                    <i class="fa-solid fa-circle-xmark text-[30px] " style="color: #F8485E;"></i>
                </button>
                <div class="flex-1 min-w-0">
                    
                    <p class="text-sm font-medium text-gray-600 truncate ">
                        {{ $item->service_name }}
                    </p>
                    <p class="text-sm text-gray-500 truncate dark:text-gray-400">
                        {{ $item->count }}
                    </p>
                </div>
                <div class="inline-flex items-center text-base font-semibold text-gray-900 ">
                    ${{ $item->service_price }}
                </div>
            </div>
        </li>
        @endforeach

    </ul>
</div>
```

**Reply by joshhanley** [reply]
@rawezhcode yeah that's correct. Does the docs say you should be using a `wire:key` in a loop?

**Reply by rawezhcode** [reply]
Yes, there is in the docs after searching I found that it is necessary to put `wire:key` in the loop.

[https://livewire.laravel.com/docs/components#adding-wirekey-to-foreach-loops](url)
![image](https://github.com/livewire/livewire/assets/53620168/cc8e3534-5c10-409b-830e-f2f864836745)

**Reply by joshhanley** [reply]
@rawezhcode ah great, thanks for checking, wanted to make sure we hadn't missed it 🙂
